compliance_checker/cf/cf.py

```diff
@@ -3698,6 +4035,286 @@ def check_cell_measures(self, ds):
 
         return ret_val
 
+    def check_grid_mapping_attr_condition(self, attr, attr_name):
+        """
+        Evaluate a condition (or series of conditions) for a particular
+        attribute. Implementation for CF-1.7.
+
+        :param attr: attribute to teset condition for
+        :param str attr_name: name of the attribute
+        :rtype tuple
+        :return two-tuple of (bool, str)
+        """
+
+        if attr_name == 'geographic_crs_name':
+            return self._evaluate_geographic_crs_name(attr)
+
+        elif attr_name == 'geoid_name': # TODO
```

Comment:
What's TODO here?  If the corresponding TODO is finished, please remove it.  If it is not, indicate what needs to be done here or implement it and remove.

Reply:
See newest commit for changes